proxy/envoy/testdata/envoy.json.golden

```diff
@@ -0,0 +1,286 @@
+{
```

Comment:
Sure, this definitely needs documenting somewhere. Let me actually write a doc on testing with general instructions and patterns for integration, goldens, etc.